### PR TITLE
[testing] ~update~ fix linting issues found by golangci-lint v1.40.1

### PR DESCRIPTION
**Pull request by thaJeztah**
Splitting this from https://github.com/moby/moby/pull/42303, which needs some work (use a custom action instead of the default one; similar to buildkit CI)

This ~updates~ fixes linting issues found by the latest golangci-lint (v1.40.1)

### CODE REVIEW

**Comment by thaJeztah** [comment]
Arf... 

```
[2021-05-31T12:16:04.349Z] #49 [golangci_lint 1/1] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg/mod     --mount=type=bind,src=hack/dockerfile/install,target=/tmp/install         PREFIX=/build /tmp/install/install.sh golangci_lint
[2021-05-31T12:16:04.349Z] #49 sha256:349e1551451e60eb111e0b092c0002ff279617974c02cedd50f0cc7c74572fef
[2021-05-31T12:16:04.349Z] #49 85.93 # github.com/quasilyte/go-ruleguard/internal/gogrep
[2021-05-31T12:16:04.349Z] #49 85.93 /tmp/tmp.hsC4laHthw/pkg/mod/github.com/quasilyte/go-ruleguard@v0.3.4/internal/gogrep/match.go:677:13: undefined: strconv.ParseComplex
[2021-05-31T12:16:04.349Z] #49 85.93 note: module requires Go 1.15
[2021-05-31T12:16:04.606Z] #49 ...
```


No longer works on Go 1.13, so we'll have to wait for https://github.com/moby/moby/pull/40353

**Comment by thaJeztah** [comment]
I'll remove the last two commits

**Comment by thaJeztah** [comment]
@AkihiroSuda @cpuguy83 @tianon ptal

File: daemon/logger/journald/read.go [hunk 1]
```diff
@@ -125,10 +125,10 @@ func (s *journald) Close() error {
 	return nil
 }
 
-// convert error code returned from a sd_journal_* function
+// CErr converts error code returned from a sd_journal_* function
 // (which returns -errno) to a string
 func CErr(ret C.int) string {
-	return C.GoString(C.strerror(C.int(-ret)))
+	return C.GoString(C.strerror(C.int(-ret))) //nolint: unconvert
```
Comment (samuelkarp) [inline daemon/logger/journald/read.go:131]: Why wouldn't we write `C.GoString(C.strerror(-ret))`?  `ret` is already a `C.int`.

File: libcontainerd/remote/client.go [hunk 2]
```diff
@@ -568,8 +568,9 @@ func (c *client) CreateCheckpoint(ctx context.Context, containerID, checkpointDi
 
 	var cpDesc *v1.Descriptor
 	for _, m := range index.Manifests {
-		if m.MediaType == images.MediaTypeContainerd1Checkpoint {
-			cpDesc = &m
+		m2 := m
```
Comment (samuelkarp) [inline libcontainerd/remote/client.go:571]: nit: `m := m` would work just as well here and wouldn't require a variable called `m2`

File: daemon/logger/journald/read.go (hunk 1 above)
Comment (thaJeztah) [reply daemon/logger/journald/read.go:131]: Hm... good call. I hate CGO, and probably wasn't sure if it was safe to remove the casting; let me update 👍

File: libcontainerd/remote/client.go (hunk 2 above)
Comment (thaJeztah) [reply libcontainerd/remote/client.go:571]: Agreed. Don't know why I didn't do so 👍  let me update

**Comment by thaJeztah** [comment]
@samuelkarp updated; ptal 👍

**Review by samuelkarp — APPROVED** [review]
LGTM

File: daemon/network.go [hunk 3]
```diff
@@ -574,7 +574,7 @@ func (daemon *Daemon) GetNetworks(filter filters.Args, config types.NetworkListC
 
 	if config.Detailed {
 		for i, n := range list {
-			np := &n
+			np := &list[i]
 			buildDetailedNetworkResources(np, idx[n.ID], config.Verbose)
```
Comment (tianon) [inline daemon/network.go:578]: Is it worth switching `n.ID` here to `np.ID` and ditching `n` completely?

**Review by tianon — APPROVED** [review]

File: daemon/network.go (hunk 3 above)
Comment (thaJeztah) [reply daemon/network.go]: Yes, makes sense; updated 👍

**Review by tianon — APPROVED** [review]

**Comment by thaJeztah** [comment]
Rebased, and resolved the conflict in the golangci-lint config file (after https://github.com/moby/moby/pull/42262 was merged)

**Comment by thaJeztah** [comment]
More flakiness in the libnetwork tests;

```
[2021-06-03T19:29:33.244Z] === Failed
[2021-06-03T19:29:33.244Z] === FAIL: libnetwork/networkdb TestNetworkDBIslands (123.95s)
```

opened https://github.com/moby/moby/issues/42459

**Comment by thaJeztah** [comment]
https://ci-next.docker.com/public/job/moby/job/PR-42445/11/execution/node/333/log/

More flakiness?

`TestFindNetworkUtil` - opened https://github.com/moby/moby/issues/42482

```
=== RUN   TestFindNetworkUtil
time="2021-06-08T08:29:35Z" level=warning msg="bridge store not initialized. kv object docker/network/v1.0/bridge/b3f5da5d7faac2508f5265a089d36408a6bc2beb5946dd80602ac6e1e4f09ef1/ is not added to the store"
time="2021-06-08T08:29:35Z" level=warning msg="Failed to update store after ipam release for network network (b3f5da5d7faac2508f5265a089d36408a6bc2beb5946dd80602ac6e1e4f09ef1): failed to update store for object type *libnetwork.network: Key not found in store"
--- FAIL: TestFindNetworkUtil (0.16s)
    api_linux_test.go:1099: Failed to delete the network: error deleting network endpoint count from store: Key not found in store
```

`TestUserChain` - opened https://github.com/moby/moby/issues/42483

```
=== RUN   TestUserChain/iptables=true,insert=true
    --- FAIL: TestUserChain/iptables=true,insert=true (0.00s)
        firewall_linux_test.go:62: assertion failed:
            --- ←
            +++ →
            {[]string}[1->?]:
            	-: "-A FORWARD -i docker0 ! -o docker0 -j ACCEPT"
            	+: <non-existent>
            {[]string}[2->?]:
            	-: "-A FORWARD -i docker0 -o docker0 -j DROP"
            	+: <non-existent>
```

**Review by samuelkarp — APPROVED** [review]
LGTM

**Pull request merged**